packages/crypto/src/public_key.ts

```diff
@@ -21,23 +20,54 @@ function str_to_key_type(keyType: string): KeyType {
     }
 }
 
-class ED25519PublicKey extends Assignable { keyType: KeyType = KeyType.ED25519; data: Uint8Array; }
-class SECP256K1PublicKey extends Assignable { keyType: KeyType = KeyType.SECP256K1; data: Uint8Array; }
+class ED25519PublicKey { keyType: KeyType = KeyType.ED25519; data: Uint8Array; }
+class SECP256K1PublicKey { keyType: KeyType = KeyType.SECP256K1; data: Uint8Array; }
+
+function resolveEnumKeyName(keyType: KeyType) {
+    switch (keyType) {
+        case KeyType.ED25519: {
+            return 'ed25519Key';
+        }
+        case KeyType.SECP256K1: {
+            return 'secp256k1Key';
+        }
+        default: {
+            throw Error(`unknown type ${keyType}`);
+        }
+    }
+}
+
+/**
+ * DUPLICATED FROM @near-js/types - REPLACE WITH IMPORTED REFERENCE AND DELETE
+ * This ends up being necessary for Wallet Selector dependencies with
+ * outdated peer dependencies and should only be temporary
+ */
+abstract class Enum {
+    abstract enum: string;
+
+    constructor(properties: any) {
+        if (Object.keys(properties).length !== 1) {
+            throw new Error('Enum can only take single value');
+        }
+        Object.keys(properties).map((key: string) => {
+            (this as any)[key] = properties[key];
+        });
+    }
+}
 
 /**
  * PublicKey representation that has type and bytes of the key.
  */
-export class PublicKey extends Assignable {
+export class PublicKey extends Enum {
```

Comment:
@gtsonevv @gagdiez I wanted to get a sanity check on this since it should have always been an `Enum` but I think worked just because `Assignable` is flexible. It now needs to extend `Enum` in order to follow the new rules around class fields and how that impacts serialization (see PR description). Doesn’t seem to cause issues but if you have concerns/insights please let me know.

Note that this also impacts `Signature` in the same way, which was similarly changed to support both ED25519 and SECP256K1 curves.